src/test/daml/Daml/Finance/Instrument/Bond/Test/InterestRateSwap.daml

```diff
@@ -0,0 +1,102 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Bond.Test.InterestRateSwap where
+
+import DA.Date
+import DA.Map qualified as M (empty, fromList)
+import DA.Set (singleton)
+import Daml.Finance.Holding.Fungible qualified as Fungible (Factory(..))
+import Daml.Finance.Instrument.Bond.Test.Util
+import Daml.Finance.Instrument.Bond.Util
+import Daml.Finance.Interface.Types.Date.Calendar
+import Daml.Finance.Interface.Types.Date.DayCount
+import Daml.Finance.Interface.Types.Date.RollConvention
+import Daml.Finance.RefData.HolidayCalendar
+import Daml.Finance.RefData.Observation (Observation(..))
+import Daml.Finance.Test.Util.Account qualified as Account (createAccount, createFactory, credit)
+import Daml.Finance.Test.Util.Common (createParties)
+import Daml.Finance.Test.Util.Instrument qualified as Instrument (originate)
+import Daml.Script
+
+-- Calculate interest rate payment on an interest rate swap, including lifecycling and creation of new instrument version
+run : Script ()
+run = script do
+  [depository, custodian, issuer, investor, calendarDataProvider, settler, provider, publicParty] <-
+    createParties ["CSD", "Custodian", "Issuer", "Investor", "Calendar Data Provider", "Settler", "Provider", "PublicParty"]
+
+  -- Account and holding factory
+  let pp = [("FactoryProvider", singleton $ singleton publicParty)]
+  accountFactoryCid <- toInterfaceContractId <$> Account.createFactory provider pp
```

Comment:
I think we can chose the `custodian` to be the account factory provider, and do away with the `provider` party.